tools/generate_release_notes.py

```diff
@@ -1,246 +1,325 @@
+https://scikit-image.org
 
-def find_author_info(commit):
-    """Return committer and author of a commit.
+"""
+    label_section_map: tuple[str, str] = (
+        (":trophy: type: Highlight", "Highlights"),
+        (":baby: type: New feature", "New Features"),
+        (":fast_forward: type: Enhancement", "Enhancements"),
+        (":chart_with_upwards_trend: type: Performance", "Performance"),
+        (":adhesive_bandage: type: Bug fix", "Bug Fixes"),
+        (":scroll: type: API", "API Changes"),
+        (":wrench: type: Maintenance", "Maintenance"),
+        (":page_facing_up: type: Documentation", "Documentation"),
+        (":robot: type: Infrastructure", "Infrastructure"),
+    )
+    ignored_user_logins: tuple[str] = ("web-flow",)
+    pr_summary_regex = re.compile(
+        r"^```release-note\s*(?P<summary>[\s\S]*?\w[\s\S]*?)\s*^```", flags=re.MULTILINE
+    )
 
-    Parameters
-    ----------
-    commit : Github commit
-        The commit to query.
+    @property
+    def intro(self):
+        return self.intro_template.format(version=self.version)
+
+    def __str__(self) -> str:
+        return "".join(self.iter_lines())
+
+    @property
+    def _prs_by_section(self) -> dict[str, set[PullRequest]]:
+        """Map pull requests to section titles.
+
+        Pull requests that lack a label which is associated with a section, are sorted
+        into a section named "Other".
+        """
+        label_section_map = {k: v for k, v in self.label_section_map}
+        prs_by_section = {
+            section_name: set() for section_name in label_section_map.values()
+        }
+        prs_by_section["Other"] = set()
+        for pr in self.pull_requests:
+            pr_labels = {label.name for label in pr.labels}
+            pr_labels = pr_labels & label_section_map.keys()
+            if not pr_labels:
+                logger.warning(
+                    "pull request %s without known section label, sorting into 'Other'",
+                    pr.html_url,
+                )
+                prs_by_section["Other"].add(pr)
+            for name in pr_labels:
+                prs_by_section[label_section_map[name]].add(pr)
+
+        return prs_by_section
+
+    def _sanitize_text(self, text: str) -> str:
+        text = text.strip()
+        text = text.replace("\r\n", " ")
+        text = text.replace("\n", " ")
+        return text
+
+    def _format_link(self, name: str, target: str) -> str:
+        return f"[{name}]({target})"
+
+    def _format_section_title(self, title: str, level: int) -> Iterable[str]:
+        yield f"{'#' * level} {title}\n"
+
+    def _parse_pull_request_summary(self, pr: PullRequest) -> str:
+        if pr.body and (match := self.pr_summary_regex.search(pr.body)):
+            summary = match["summary"]
+        else:
+            logger.debug("falling back to title for %s", pr.html_url)
+            summary = pr.title
+        summary = self._sanitize_text(summary)
+        return summary
```

Comment:
This method handles the parsing for the PR summary (I call it that for lack of a better term right now). Note that `self._sanitize_text(summary)` is replacing `\r\n` and `\n` with a space naively. I'm sure there are edge cases the script is not yet handling well. :sweat_smile: